### PR TITLE
Use Spark's `Utils.getContextOrSparkClassLoader` to load Shims [databricks]

### LINKED ISSUE #3851: [BUG] ShimLoader.updateSparkClassLoader fails with openjdk Java11
**Describe the bug**
Originally reported by @abellina 

```
21/10/18 23:18:41 INFO ShimLoader: Updating spark classloader jdk.internal.loader.ClassLoaders$AppClassLoader@5ffd2b27 with the URLs:
 jar:file:/home/gshegalov/gits/NVIDIA/spark-rapids/dist/target/rapids-4-spark_2.12-21.12.0-SNAPSHOT.jar!/spark3xx-common/, jar:file:/
home/gshegalov/gits/NVIDIA/spark-rapids/dist/target/rapids-4-spark_2.12-21.12.0-SNAPSHOT.jar!/spark320/
21/10/18 23:18:41 ERROR CoarseGrainedExecutorBackend: Executor self-exiting due to : Unable to create executor due to No such method:
 addURL() on object: jdk.internal.loader.ClassLoaders$AppClassLoader
java.lang.NoSuchMethodException: No such method: addURL() on object: jdk.internal.loader.ClassLoaders$AppClassLoader
        at org.apache.commons.lang3.reflect.MethodUtils.invokeMethod(MethodUtils.java:231)
        at org.apache.commons.lang3.reflect.MethodUtils.invokeMethod(MethodUtils.java:184)
        at com.nvidia.spark.rapids.ShimLoader$.$anonfun$updateSparkClassLoader$3(ShimLoader.scala:216)
```

**Steps/Code to reproduce bug**

Set `JAVA_HOME` environment  to an openjdk java11 location and run code with the default `spark.rapids.force.caller.classloader=true`. 

Note this issue does not reproduce with the Microsoft Build of OpenJDK 11

**Expected behavior**
It should work.

- short term just add another [method name](https://github.com/openjdk/jdk/blob/6765f902505fbdd02f25b599f942437cd805cad1/src/java.base/share/classes/jdk/internal/loader/ClassLoaders.java#L203) to try in the reflection calls, although JDK module encapsulation [will not make it easy ](https://stackoverflow.com/questions/41265266/how-to-solve-inaccessibleobjectexception-unable-to-make-member-accessible-m) 
- long term potentially #3803

**Environment details (please complete the following information)**
local dev environment 

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by gerashegalov**
Fixes #3851

Spark loads external [datasources][1] using `Utils.getContextOrSparkClassLoader`

Trampoline to `Utils.getContextOrSparkClassLoader`  to make our current code work with external sources, and to unblock JDK9+ 

[1]:  https://github.com/apache/spark/blob/b63674ea5f746306a96ab8c39c23a230a6cb9566/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSource.scala#L652 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
current premerge status:
```
2 failed, 562 passed, 23 skipped, 4 xfailed in 619.26s (0:10:19) 

An error occurred while calling z:com.nvidia.spark.rapids.ExecutionPlanCaptureCallback.assertDidFallBack. : java.lang.NoClassDefFoundError: com/nvidia/spark/rapids/shims/SparkShimImpl$ 
```

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
Many more tests passed.
Next issues `-k 'test_ts_read_fails_datetime_legacy[-reader_confs0-LEGACY-INT96-Timestamp]'`
```
13:11:39  E                   Caused by: java.lang.UnsatisfiedLinkError: com.nvidia.spark.rapids.jni.ParquetFooter.readAndFilter(JJJJ[Ljava/lang/String;[IIZ)J
13:11:39  E                   	at com.nvidia.spark.rapids.jni.ParquetFooter.readAndFilter(Native Method)
13:11:39  E                   	at com.nvidia.spark.rapids.jni.ParquetFooter.readAndFilter(ParquetFooter.java:90)
13:11:39  E                   	at com.nvidia.spark.rapids.GpuParquetFileFilterHandler.$anonfun$readAndFilterFooter$5(GpuParquetScan.scala:594)
13:11:39  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
13:11:39  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
13:11:39  E                   	at com.nvidia.spark.rapids.GpuParquetFileFilterHandler.withResource(GpuParquetScan.scala:460)
13:11:39  E                   	at com.nvidia.spark.rapids.GpuParquetFileFilterHandler.$anonfun$readAndFilterFooter$4(GpuParquetScan.scala:586)
```

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
retrigger workflows

**Review by jlowe — APPROVED** [review]
Loving this.

File: spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala [hunk 1]
```diff
@@ -1402,14 +1402,6 @@ object RapidsConf {
     .booleanConf
     .createWithDefault(true)
 
-  val FORCE_SHIMCALLER_CLASSLOADER = conf("spark.rapids.force.caller.classloader")
```
Comment (revans2) [inline spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala:1405]: Does anything else in spark2 use this?

**Review by revans2 — DISMISSED** [review]

File: spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala (hunk 1 above)
Comment (gerashegalov) [reply spark2-sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala:1482]: This is merely due to spark2 cloning RapidsConf source. We verified there is no actual use of spark.rapids.force.caller.classloader in the spark 2 explain support.

**Comment by gerashegalov** [comment]
Upmerged and removed newer references to force.caller.classloader.
 
Ran iceberg tests locally without issues using the local and standalone modes.

**Comment by gerashegalov** [comment]
build

**Review by jlowe — APPROVED** [review]

**Pull request merged**